pkgs/by-name/gs/gsad/package.nix

```diff
@@ -0,0 +1,146 @@
+{
+  cmake,
+  doxygen,
+  fetchFromGitHub,
+  gettext,
+  git,
+  glib,
+  gnutar,
+  gnutls,
+  gvm-libs,
+  lib,
+  libidn2,
+  libgcrypt,
+  libmicrohttpd,
+  libselinux,
+  libsepol,
+  libtasn1,
+  libxml2,
+  libxslt,
+  p11-kit,
+  pcre2,
+  pkg-config,
+  python3Packages,
+  stdenv,
+  util-linux,
+  xmltoman,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gsad";
+  version = "22.10.0";
+
+  # web app needed to start web server
+  src = fetchFromGitHub {
+    owner = "greenbone";
+    repo = "gsa";
+    rev = "refs/tags/v23.1.1";
+    hash = "sha256-yA+B6fh8C07tGEoy/+FPvcvx3D1OfPig0Eoqv2oJfsA=";
+  };
+
+  # web server
+  src2 = fetchFromGitHub {
```

Comment:
The names are misleading.

- The package that uses the same package version should be `src`.

- Name the web app source as `webAppSource`. 
  - externalize it to a `let ... in`, because this is not meaning to the derivation.